wrapper/src/main/java/software/amazon/jdbc/plugin/AuroraConnectionTrackerPlugin.java

```diff
@@ -128,9 +129,11 @@ public <T, E extends Exception> T execute(final Class<T> resultClass, final Clas
 
     } catch (final Exception e) {
       if (e instanceof FailoverSQLException) {
-        tracker.invalidateAllConnections(this.currentWriter);
-        tracker.logOpenedConnections();
-        this.needUpdateCurrentWriter = true;
+        if (!Objects.equals(this.getWriter(this.pluginService.getHosts()), this.currentWriter)) {
```

Comment:
Not sure if we should use Objects.equals - don't we have another way of checking the equality here?

Reply:
What this condition does?

Reply:
Yes, HostSpec implements equals so no need to use Object.equals()

Reply:
> What this condition does?

In case the cluster gets rebooted without failover.

> Not sure if we should use Objects.equals - don't we have another way of checking the equality here?

Is there any concerns here? like efficiency or something?

`Objects.equals` is used because it handles `null` value. `HostSpec.equals()` may throw Nullpointer if the variable is `Null`.

Reply:
No it won't return a NullPointerException `null instanceof HostSpec` returns false